File: Data Structures & Algorithms/subsets-ii/submission-0.cpp
class Solution {
public:
    vector<vector<int>> res;
    vector<int> path;

    void backtrack(vector<int>& nums, int start) {
        res.push_back(path);

        for (int i = start; i < nums.size(); i++) {
            //Bỏ qua các giá trị bằng nhau tại cùng cấp độ
            if (i > start && nums[i] == nums[i -1]) {
                continue;
            }
            path.push_back(nums[i]);
            backtrack(nums, i + 1); //mở rộng
            path.pop_back();
        }
    }
    vector<vector<int>> subsetsWithDup(vector<int>& nums) {
        sort(nums.begin(), nums.end());
        backtrack(nums, 0);
        return res;
    }
};

File: Data Structures & Algorithms/target-sum/submission-0.py
class Solution:
    def findTargetSumWays(self, nums: List[int], target: int) -> int:
        def backtrack(i, curr_sum):
            if i == len(nums):
                return 1 if curr_sum == target else 0
            
            return (backtrack(i + 1, nums[i] + curr_sum) +
                    backtrack(i + 1, curr_sum - nums[i]))
        return backtrack(0, 0)